src/utils/consistency/mod.rs

```diff
@@ -1,59 +1,300 @@
+/// apply any changes that we find in the index but not our database.
+///
+/// Differences that we check for, and the activivies:
+/// * release in index, but not our DB => queue a build for this release.
+/// * crate in index, but not in our DB => queue builds for all versions of that crate.
+/// * release in DB, but not in the index => delete the release from our DB & storage.
+/// * crate in our DB, but not in the index => delete the whole crate from our DB & storage.
+/// * different yank-state between DB & Index => update the yank-state in our DB
+///
+/// Even when activities fail, the command can just be re-run. While the diff calculation will
+/// be repeated, we won't we-execute fixing activities.
+pub fn run_check(ctx: &dyn Context, dry_run: bool) -> Result<()> {
+    let mut conn = ctx.pool()?.get()?;
+    let index = ctx.index()?;
 
     info!("Loading data from database...");
-    let timer = std::time::Instant::now();
-    let db_data =
-        self::db::load(conn).context("Loading crate data from database for consistency check")?;
-    info!("...loaded in {:?}", timer.elapsed());
+    let db_data = db::load(&mut conn, &*ctx.config()?)
+        .context("Loading crate data from database for consistency check")?;
 
     tracing::info!("Loading data from index...");
-    let timer = std::time::Instant::now();
     let index_data =
-        self::index::load(index).context("Loading crate data from index for consistency check")?;
-    info!("...loaded in {:?}", timer.elapsed());
-
-    let diff = db_data.diff(index_data);
-
-    for krate in diff.crates {
-        match krate {
-            Diff::Both(name, diff) => {
-                for release in diff.releases {
-                    match release {
-                        Diff::Both(_, _) => {}
-                        Diff::Left(version, _) => {
-                            println!("Release in db not in index: {} {}", name, version);
-                        }
-                        Diff::Right(version, _) => {
-                            println!("Release in index not in db: {} {}", name, version);
-                        }
+        index::load(&index).context("Loading crate data from index for consistency check")?;
+
+    let diff = diff::calculcate_diff(db_data.iter(), index_data.iter());
+    let result = handle_diff(ctx, diff.iter(), dry_run)?;
+
+    println!("============");
+    println!("SUMMARY");
+    println!("============");
+    println!("difference found:");
+    for (key, count) in diff.iter().counts_by(|el| match el {
+        diff::Difference::CrateNotInIndex(_) => "CrateNotInIndex",
+        diff::Difference::CrateNotInDb(_, _) => "CrateNotInDb",
+        diff::Difference::ReleaseNotInIndex(_, _) => "ReleaseNotInIndex",
+        diff::Difference::ReleaseNotInDb(_, _) => "ReleaseNotInDb",
+        diff::Difference::ReleaseYank(_, _, _) => "ReleaseYank",
+    }) {
+        println!("{} => {}", key, count);
```

Comment:
```suggestion
        println!("{:17} => {:4}", key, count);
```

Vertically aligning these would be nice